File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.go
func twoSum(numbers []int, target int) []int {
    n := len(numbers)
    i := 0
    j := n-1
    for _, _ = range numbers{
        if numbers[i] + numbers[j] > target{
            j--
        }else if numbers[i] + numbers[j] < target{
            i++
        }else{
            return []int{i+1,j+1}
        }
    }
    return []int{}

}

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for i in range(len(matrix)):
            row = matrix[i]
            start, end = 0, len(row) - 1
            x=row[start]
            y=row[end]
            if target >= x and target <= y:
                mid = start + (end-start)//2
                while start<=end and end>=0:
                    if target > row[mid]:
                        start = mid + 1
                    elif target < row[mid]:
                        end = mid - 1
                    elif target == row[mid]:
                        return True
                    else:
                        return False
                    mid = start + (end-start)//2
            else:
                continue
        return False

